File: library/src/main/java/cz/developer/library/DeveloperConfig.java
package cz.developer.library;

import cz.developer.library.ui.image.ImageAdapter;
import cz.developer.library.ui.network.INetworkAdapter;
import cz.developer.library.ui.switchs.ISwitchInterface;

/**
 * Created by Administrator on 2016/11/7.
 */

public class DeveloperConfig {
    public ISwitchInterface switchConfig;
    public INetworkAdapter networkAdapter;
    public ImageAdapter imageAdapter;
    public String channel;
    private DeveloperConfig(){
    }

    public static class Builder{
        private DeveloperConfig config;
        public Builder() {
            this.config = new DeveloperConfig();
        }

        public Builder setSwitchInterface(ISwitchInterface switchConfig){
            this.config.switchConfig =switchConfig;
            return this;
        }

        public Builder setNetworkAdapter(INetworkAdapter adapter){
            this.config.networkAdapter=adapter;
            return this;
        }

        public Builder setImageAdapter(ImageAdapter adapter){
            this.config.imageAdapter=adapter;
            return this;
        }

        public Builder setChannel(String channel){
            this.config.channel=channel;
            return this;
        }


        public DeveloperConfig build(){
            return this.config;
        }
    }
}

File: library/src/main/java/cz/developer/library/DeveloperConfig.kt
package cz.developer.library

/**
 * Created by cz on 2016/11/7.
 */

class DeveloperConfig {
    //启用视图
    var hierarchy=false
    //配置渠道
    var channel: String?=null
    //切换控制条目
    internal var switchItem:SwitchItem?=null
    //状态切换配置
    fun switch(config:SwitchItem.()->Unit){
        switchItem=SwitchItem().apply(config)
    }
    class SwitchItem {
        internal val items= mutableListOf<Item>()
        //条目是否选中
        internal var itemChecked:((String,Boolean)->Unit)?=null
        //获取附加集
        fun item(key:String?=null,desc:String?=null){
            items.add(Item(key,desc))
        }

        fun onItemChecked(action:(String,Boolean)->Unit){
            this.itemChecked=action
        }

        class Item(var key:String?=null, var desc:String?=null)


    }

}
